scripts/cmake/vcpkg_copy_tool_dependencies.cmake

```
@@ -26,9 +48,11 @@ function(vcpkg_copy_tool_dependencies tool_dir)
     endif()
 
     if(VCPKG_TARGET_IS_WINDOWS)
-        find_program(Z_VCPKG_POWERSHELL_CORE pwsh)
-        if (NOT Z_VCPKG_POWERSHELL_CORE)
-            message(FATAL_ERROR "Could not find PowerShell Core; please open an issue to report this.")
+        if (CMAKE_MAJOR_VERSION VERSION_LESS 3 OR CMAKE_MINOR_VERSION VERSION_LESS 16)
```

broken with eventual cmake 4.0

---

There should be no need to test the individual version components separately. Operators like `VERSION_LESS` handle full version numbers:

```suggestion
        if(CMAKE_VERSION VERSION_LESS 3.16)
```

---

`CMAKE_VERSION` is only exists after cmake 2.6.3, see https://cmake.org/cmake/help/latest/variable/CMAKE_VERSION.html

---

are we supporting such an old CMake version in downstream projects? I don't think so!
also this script runs in internal cmake, which means we can use the latest and greatest, right?

---

> are we supporting such an old CMake version in downstream projects? I don't think so! also this script runs in internal cmake, which means we can use the latest and greatest, right?

Good question! Therefore, we must choose one of the following ways:
1. Always use cmake from vcpkg, which will go against `VCPKG_FORCE_BINARIES`.
2. Keep the original code and enable the new code when the current cmake version is higher than the version that can support this feature.

---

Currently, we are working on directly incorporating this function into vcpkg-tool, including parsing PE, elf and mach-O formats.

---

> Good question!

No, it was a rhetorical question. vcpkg requires at least CMake 3.7 in user projects, and script may rely on the version which is downloaded by vcpkg.

---

Cf. https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/cmake-guidelines.md#cmake-versions-to-require